Metrics: fetch,display&test data

tonnymuchui opened the pull request:
### This is a metrics website application for digital currency. Users can get information about a related crypto coin. I fetched data from an Api `coincap`.
- Used React components.
- Used React Router.
- Used React life cycle methods.
- Used store, actions, and reducers.
- Create unit tests for pure functions.
- Retrieved data from an API and stored in the Redux store.
- Live Link: https://bespoke-cassata-f52719.netlify.app/
- Video Link One: https://www.loom.com/share/4907b20fd40b4d4b976e2019f3b77351
- Video  Link Two: https://www.loom.com/share/13baf71328c8402f95b710d2badd6227

tonnymuchui committed:
Add linters

tonnymuchui committed:
Configure linters

tonnymuchui committed:
Add configure store

tonnymuchui committed:
Add details redux

tonnymuchui committed:
Add list redux

tonnymuchui committed:
Add Coin Component

tonnymuchui committed:
Add Details page

tonnymuchui committed:
Add Details page styling

tonnymuchui committed:
Add Home page

tonnymuchui committed:
Add Home page styling

tonnymuchui committed:
Add providers

tonnymuchui committed:
Add Routers

tonnymuchui committed:
Add styling

tonnymuchui committed:
Add redux

tonnymuchui committed:
Add test snapshot

tonnymuchui committed:
Add test coin

tonnymuchui committed:
Add test home

tonnymuchui committed:
Add test details

tonnymuchui committed:
Update dependencies

tonnymuchui committed:
Update test

tonnymuchui committed:
Add assets

tonnymuchui committed:
Add chart component

tonnymuchui committed:
Add header component

tonnymuchui committed:
Update coin component

tonnymuchui committed:
Update page component

tonnymuchui committed:
Add styling for header

tonnymuchui committed:
Add styling for chart

tonnymuchui committed:
Add styling for coin

tonnymuchui committed:
Add styling for details

tonnymuchui committed:
Add styling for home

tonnymuchui committed:
Add styling for app

tonnymuchui committed:
delete app.test

tonnymuchui committed:
Update packages

tonnymuchui committed:
fix linters error

tonnymuchui committed:
Add readme

tonnymuchui committed:
Merge pull request #3 from tonnymuchui/setup-and-Api

Setup and api

tonnymuchui committed:
Merge pull request #4 from tonnymuchui/testing

Testing

tonnymuchui committed:
Update Readme

Owusu-Desmond commented on README.md:
```diff
@@ -1,70 +1,235 @@
+> Mention all of the collaborators of this project.
+
+👤 **Author1**
+
```
- [ ] Please since you are the only contributor to this project you can singularize the `Authors`  to `Author` and replace `Author1` with your name. :smile:

Owusu-Desmond commented on README.md:
```diff
@@ -1,70 +1,235 @@
+## 🚀 Live Demo <a name="live-demo"></a>
 
-### Analyzing the Bundle Size
+- [LiveDemo](https://bespoke-cassata-f52719.netlify.app/)
 
```
- [ ]  Please make sure that your presentation video contains a highlight about a piece of the code that you find interesting or are proud of making. smile
  
  #### As requested in the project requirement.
  ![image](https://user-images.githubusercontent.com/92303891/218274520-501483bb-4c76-4292-a6b1-840da7664a09.png)

Owusu-Desmond commented on README.md:
```diff
@@ -1,70 +1,235 @@
+
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+> Give credit to everyone who inspired your codebase.
+
+I would like to thank Microverse for their contributions.
+
+<!-- <p align="right">(<a href="#readme-top">back to top</a>)</p> -->
```
- [ ] Please make sure that you give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/). :blush:

Owusu-Desmond commented on src/_tests_/Coin.test.js:
```diff
@@ -0,0 +1,25 @@
+import { render, screen } from '@testing-library/react';
+import Coin from '../components/Coin';
+import '@testing-library/jest-dom/extend-expect';
+
+const data = {
+  name: 'name',
+  symbol: 'symbol',
+  id: '1',
+  price: '1000',
+};
+describe('coin component', () => {
+  it('elements', () => {
+    render(<Coin
+      name={data.name}
+      symbol={data.symbol}
+      id={data.id}
+      price={data.price}
+      handleDetails={(res) => res}
+    />);
+    expect(screen.getByRole('button')).toBeInTheDocument();
+    expect(screen.getByText('name')).toBeInTheDocument();
+    expect(screen.getByText('symbol')).toBeInTheDocument();
+    expect(screen.getByText(/1000/i)).toBeInTheDocument();
+  });
```
- [ ] Nice to test your project using `testing-library/jest-dom` but you still have some of your tests failing so please make sure that you fix these failing tests. :pray: 

![Screenshot from 2023-02-11 18-31-40](https://user-images.githubusercontent.com/92303891/218275532-2064cc6c-15e8-42a7-b988-f0062e702d85.png)

Owusu-Desmond reviewed (changes requested):
Hi @tonnymuchui , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://media2.giphy.com/media/Jq824R93JsLwZCaiSL/giphy.gif"/>

## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

tonnymuchui committed:
Update test

tonnymuchui committed:
Update

tonnymuchui committed:
Update test

tonnymuchui committed:
fix linters problem

tonnymuchui committed:
Add acknowledgment

tonnymuchui committed:
Update configurestore

tonnymuchui committed:
Update coin component

tonnymuchui committed:
Update coin test

tonnymuchui committed:
Update packages

tonnymuchui committed:
fix linters error

tonnymuchui committed:
Add video two url

tonnymuchui commented:
- Fixed Readme author contributor :heavy_check_mark:
- Added video presentation for highlighting code.(used my GitHub to highlight code because new version of Ubuntu has problem with presentation)  :heavy_check_mark:
- Added Acknowledgement. :heavy_check_mark:
- Fixed test. :heavy_check_mark:

jerryowusu commented on src/pages/Details.js:
```diff
@@ -0,0 +1,72 @@
+          <span>VWAP(24Hr) :</span>
+          <span>{details.vwap24Hr}</span>
+        </li>
+      </ul>
```
Nice and beautiful design so far, however, as stated in the project requirements, you are suppose to follow the design guidelines including: 

- [x] Colors (select one main color for your website).
- [x] Typography: font face, size and weight.
- [x] Layout: composition and space between elements.

There are some differences between yours and that of the actual design which you will need to fix.

1. Kindly note that your design is missing the small arrow to the right of each card.

### Your design 

![Screenshot from 2023-02-18 12-46-03](https://user-images.githubusercontent.com/65233860/219867087-394fcc6b-08a6-42c6-8ae0-ab673831327f.png)


### Actual design 

![Screenshot from 2023-02-18 12-57-19](https://user-images.githubusercontent.com/65233860/219867632-af2ec6aa-199d-465d-b426-01a2db4e6bae.png)

jerryowusu commented on src/pages/Home.js:
```diff
@@ -0,0 +1,67 @@
+      <Header path="/" />
+      <Chart title1="Crypto" title2="Currencies" count={`${coins.length} Coins`} />
+
+      <div>
+
+        <input
+          className="coins-abt"
+          type="text"
+          value={search.get('filter') || ''}
+          placeholder="Search by Crypto name  eg : BNB"
+          onChange={(e) => {
+            const filter = e.target.value;
+            if (filter) {
+              setSearch({ filter });
+            } else {
+              setSearch({});
+            }
+          }}
+        />
+        <div className="coin-card-1">
+          {coins
+            .filter((coin) => {
+              const filter = search.get('filter');
+              if (!filter) return true;
+              const name = coin.name.toLowerCase();
+              return name.startsWith(filter.toLowerCase());
+            })
+            .map((coin) => (
+              <Coin
+                key={coin.id}
+                name={coin.name}
+                symbol={coin.symbol}
+                id={coin.id}
+                price={coin.priceUsd}
+                handleDetails={handleDetails}
+              />
+            ))}
+        </div>
```
Nice and beautiful design so far, however, as stated in the project requirements, you are suppose to follow the design guidelines including: 

- [x] Colors (select one main color for your website).
- [x] Typography: font face, size and weight.
- [x] Layout: composition and space between elements.

There are some differences between yours and that of the actual design which you will need to fix.

1. The background of each card alternates between light and darker color along the column forming a checkered layout.
2. The position of the small arrow in the circle is at the top right corner and not the top middle. 

### Your design 


![Screenshot from 2023-02-18 12-46-12](https://user-images.githubusercontent.com/65233860/219867867-fc4cc934-8804-432a-96fa-52f8666cf104.png)


### The actual design 


![Screenshot from 2023-02-18 12-57-13](https://user-images.githubusercontent.com/65233860/219868038-e10adb4d-67a0-4e6b-a929-f67c8af1d6d5.png)

jerryowusu reviewed (changes requested):
Hi @tonnymuchui , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

![almost-there-reach](https://user-images.githubusercontent.com/65233860/219866804-da3c52ec-389b-4719-9174-183aa5b835e3.gif)


## Highlights
Good job so far :+1: 
No linter errors :+1: 

## Required Changes ♻️

- [ ] Well done for adding test for the coin component. However, as part of the project requirement, you are suppose to add tests for several components, so testing for only one component is not enough. You are also required to test for some of the pure functions in the redux code. :pray: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

tonnymuchui committed:
Update styling

tonnymuchui committed:
Update component

tonnymuchui committed:
Update styling

tonnymuchui commented:
- Updated the position of the small arrow :heavy_check_mark: 
- Fixed all the styling required :heavy_check_mark:

sumairq reviewed (approved):
### Hi @tonnymuchui , 

<img src="https://media3.giphy.com/media/BPJmthQ3YRwD6QqcVD/giphy.gif"/>

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
